ffmpeg/dag.py

```diff
@@ -69,31 +70,38 @@ def incoming_edge_map(self):
         raise NotImplementedError()
 
 
-DagEdge = namedtuple('DagEdge', ['downstream_node', 'downstream_label', 'upstream_node', 'upstream_label'])
+DagEdge = namedtuple('DagEdge', ['downstream_node', 'downstream_label', 'upstream_node', 'upstream_label', 'upstream_selector'])
 
 
 def get_incoming_edges(downstream_node, incoming_edge_map):
     edges = []
-    for downstream_label, (upstream_node, upstream_label) in list(incoming_edge_map.items()):
-        edges += [DagEdge(downstream_node, downstream_label, upstream_node, upstream_label)]
+    # downstream_label, (upstream_node, upstream_label) in [(i[0], i[1][:2]) for i in self.incoming_edge_map.items()]
+    for downstream_label, upstream_info in [(i[0], i[1]) for i in incoming_edge_map.items()]:
```

Comment:
This line is magical and overly complex. What do `i[0]` and `i[1]` mean? Can we split this out into something that's more obvious to understand?